Use ament_python to build packages rather than ament_cmake

EricGallimore opened the pull request:
**Public API Changes**
None


**Description**
Change the build system for these packages from `ament_cmake` to `ament_python`.  This is the suggested way to build Python packages for ROS2, and is needed for them to work under Windows.
- `rosapi`
- `rosbridge_server`

The build system for `rosbridge_library` is left alone in this PR, since it doesn't have any scripts that need to be executable.  However, it could also be converted for consistency.

This should resolve #914 and resolve #921.

I've verified that this test launch file works on Windows with Jazzy:
```
ros2 launch rosbridge_server rosbridge_websocket_launch.xml
```

EricGallimore committed:
Changed type hints in ros_loader.py to use imports from Typing, restoring compatibility with Python <3.9.

EricGallimore committed:
Reordered imports alphabetically to fix isort linting CI error.

EricGallimore committed:
Change rosapi package to use ament_python rather than ament_cmake.

EricGallimore committed:
Change rosapi package to use ament_python rather than ament_cmake.

EricGallimore committed:
Change rosbridge_server to use ament_python rather than ament_cmake.

EricGallimore committed:
Change rosbridge_server to use ament_python rather than ament_cmake.

EricGallimore committed:
Fix setup.py formatting.

EricGallimore committed:
Fix setup.py formatting.

EricGallimore committed:
Run pre-commit and commit fixes.  (Not sure why this is needed, since some of these files aren't touched in this branch.)

EricGallimore committed:
Run pre-commit and commit fixes.  (Not sure why this is needed, since some of these files aren't touched in this branch.)

EricGallimore committed:
Revert inadvertent changes to rosbridge_websocket symlink.

EricGallimore committed:
Revert inadvertent changes to rosbridge_websocket symlink.

EricGallimore committed:
Merge branch '937-type-string-dependency' into ament_python

EricGallimore committed:
Fix launch files in rosbridge_server.

EricGallimore committed:
Fix launch files in rosbridge_server.

EricGallimore committed:
Fix setup.cfg paths.

EricGallimore committed:
Fix setup.cfg paths.

EricGallimore committed:
Fix setup.py data_files.

EricGallimore committed:
Fix setup.py data_files.

EricGallimore committed:
Moved rosapi_node and rosbridge_websocket.py into respective packages.

EricGallimore committed:
Moved rosapi_node and rosbridge_websocket.py into respective packages.

EricGallimore committed:
Add resource files.

EricGallimore committed:
Add resource files.

EricGallimore committed:
Reorganize package to follow ROS2 python convention (and resolve issue with find_packages)

EricGallimore committed:
Reorganize package to follow ROS2 python convention (and resolve issue with find_packages)

EricGallimore committed:
Run pre-commit and fix linting errors.

EricGallimore committed:
Run pre-commit and fix linting errors.

EricGallimore commented:
This PR includes the changes from #938, since they were needed to run under Windows for testing.  That PR should probably be merged first.

sea-bass committed:
Merge branch 'ros2' into ament_python

sea-bass commented on rosapi/rosapi/rosapi_node.py:
This file seems to have some permissions changes. Can you fix this to make sure it's executable?

sea-bass commented on rosbridge_server/rosbridge_server/rosbridge_websocket.py:
Same here

sea-bass commented on rosbridge_server/rosbridge_server/websocket_handler.py:
And here

sea-bass reviewed (changes requested):
Changes look good, but seems some executable files became not executable in the process. Please take a look.

bmartin427 committed:
rosapi: Don't start parameter services that aren't spun (#943) (#944)

bmartin427 committed:
rosbridge_websocket: Stop websocket server if ROS shuts down (#945) (#946)

Signed-off-by: Brad Martin <brad.martin@merlinlabs.com>

bmartin427 committed:
Avoid stale subscription when unsubscribing during resubscription (#947) (#948)

If a client asks to subscribe to the same topic more than once, then disconnects, only a
pending 'new' subscription may get deleted, leaving the 'old' subscription active
indefinitely.

Signed-off-by: Brad Martin <brad.martin@merlinlabs.com>

ManuETR committed:
docs: update ROSBRIDGE_PROTOCOL.md - correcting indentation (#949)

Add whitespace in front of all ROS action operations -- for correct indentation in Markdown

dnikitop committed:
fix: update new subs with dds from publisher (#940)

EricGallimore committed:
Fixed executable bit on scripts.

EricGallimore committed:
Fixed executable bit on scripts.

EricGallimore committed:
Merge branch 'ament_python' of https://github.com/EricGallimore/rosbridge_suite into ament_python

EricGallimore commented:
> Changes look good, but seems some executable files became not executable in the process. Please take a look.

These should be fixed, thanks.

sea-bass commented:
There are also some merge conflicts to address (sorry, I merged a bunch of PRs recently)

EzraBrooks committed:
Fix length-limited list types (#840)

EricGallimore committed:
Merge remote-tracking branch 'origin/ament_python' into ament_python

EricGallimore commented:
> There are also some merge conflicts to address (sorry, I merged a bunch of PRs recently)

Just rebased and merged.  Hopefully this does it.

sea-bass commented:
Weirdly enough, it also shows a bunch of other changes which seem to undo/repeat some of the small PRs that recently went in ... did you rebase on the latest version of the upstream `ros2` branch, or the latest that you had checked out?
